### PR TITLE
Add helm chart to release processes in v3.17

### PR DESCRIPTION
**Pull request by lwr20**
## Description

Copy helm chart release machinery from master, and the post-release test to check it worked.

## Todos

- [ ] Tests
- [ ] Documentation
- [ ] Release note

## Release Note

```release-note
None required
```

### CODE REVIEW

File: Makefile [hunk 1]
```diff
@@ -305,9 +306,9 @@ update_canonical_urls:
 
 ## Tags and builds a release from start to finish.
 release: release-prereqs
-	$(MAKE) release-tag
-	$(MAKE) release-build
-	$(MAKE) release-verify
+	$(MAKE) RELEASE_CHART=true release-tag
```
Comment (ozdanborne) [inline Makefile:309]: What is the purpose of `RELEASE_CHART`? Is it similar to the `CONFIRM` var we use on other projects?

File: release-scripts/tests/test_helm_artifacts.py [hunk 2]
```diff
@@ -0,0 +1,31 @@
...
+with open("%s/_data/versions.yml" % DOCS_PATH) as f:
+    versions = yaml.safe_load(f)
+    RELEASE_VERSION = versions[0]["title"]
+    print("[INFO] using _data/versions.yaml, discovered version: %s" % RELEASE_VERSION)
+
+# Helm version is the calico/node version.
+version = versions[0]["components"]["calico/node"]["version"]
+chart_version = versions[0]["chart"]["version"]
+print("[INFO] using calico/node version for Helm artifacts: %s" % version)
+print("[INFO] using chart version for Helm artifact: %s" % chart_version)
```
Comment (ozdanborne) [inline release-scripts/tests/test_helm_artifacts.py:23]: Seems like this functionality is pretty similar to the `$(YAML_CMD) read - '[0].chart.version'` done in the Makefile. Should we just pass it in as an arg instead?

File: release-scripts/tests/test_helm_artifacts.py [hunk 3]
```diff
@@ -0,0 +1,31 @@
+import os
+import yaml
+import requests
+
```
Comment (ozdanborne) [inline release-scripts/tests/test_helm_artifacts.py:4]: I don't see this script being called from anywhere. Is it documented to run it somewhere?

**Review by ozdanborne — APPROVED** [review]
I realize this is just a cherry-pick so I'm approving, but had a few questions as to why it was done this way

File: Makefile (hunk 1 above)
Comment (lwr20) [reply Makefile:309]: From line 67: 
```
# the presence of RELEASE_CHART indicates we're trying to cut an official chart release.
```

File: release-scripts/tests/test_helm_artifacts.py (hunk 2 above)
Comment (lwr20) [reply release-scripts/tests/test_helm_artifacts.py:23]: I guess we could, but that same argument applies to basically all the other vars in `versions.yml`.  This is at least consistent with the other post-release tests.

File: release-scripts/tests/test_helm_artifacts.py (hunk 3 above)
Comment (lwr20) [reply release-scripts/tests/test_helm_artifacts.py:4]: The post-release tests are run by 
```
make release-test
```
which is described in the "Verifying the release" section of `RELEASING.md`.

File: Makefile (hunk 1 above)
Comment (ozdanborne) [reply Makefile:309]: OK. are you sure you don't want to use `CONFIRM=true` as it's more common across our makefile machinery?

Comment (caseydavenport) [reply Makefile:309]: I don't think this is the same thing as CONFIRM. This is "do we want to publish an official helm chart", not "are you sue you want to make a release"

**Pull request merged**